test/libsolidity/AnalysisFramework.cpp

```diff
@@ -191,3 +185,24 @@ FunctionTypePointer AnalysisFramework::retrieveFunctionBySignature(
 {
 	return _contract.interfaceFunctions()[util::selectorFromSignatureH32(_signature)];
 }
+
+std::string AnalysisFramework::withPreamble(std::string const& _sourceCode)
```

Comment:
A bit of a nit: can you please continue using the existing style when it comes to prepending `std::` namespace, i.e. `string` instead of `std::string`, since it makes it more annoying for me to change it later on.

Reply:
Sure. No problem. I can change that here too.

I'm curious why it would be a problem though. If any thing, I'd expect it to be more heplful than detrimental. I'm touching the line anyway so it's not like we can avoid a conflict here. And you need to account for stuff already having `std::` anyway.

Reply:
The less instances of `std::` in the file, the easier it gets to automatically replaces all occurrences of e.g. `string` in the file. If you already have `std::string`, then the auto replace will turn this into `std::std::string`, which is the annoying part. And before you suggest usage of ` string` -> ` std::string` transform, there's also use cases such as `vector<string>` where this approach won't work. I've also been unable to find a tool to automatically detect all standard library containers and functions and auto insert `std::` namespace. Of course, even if you were to merge this, it's not a massive problem, I'm just noting it so you know for future reference.

Reply:
`std::` removed.

But having to deal with that sounds extremely tedious. Are you using autoreplace in your editor or something? This sounds like it would be very simple to do with a one-liner with much fewer false-positives.

This seems to do the job well enough:

```bash
find . -name *.cpp | xargs \
    sed -i -e 's#\0(^\|[^:a-zA-Z0-9_/<]\)string\($\|[^a-zA-Z_/>]\)#\1std::string\2#g'
```

The main thing it gets wrong are variables named `string` (we do seem to have some in macros) and the word `string` in comments. If you really wanted to, you could probably adapt the command to deal with the latter (at least for single-line comments).

Reply:
I've used exactly what you suggested initially, and found the autoreplace in the editor to be the quickest approach (counter intuitively), since in the end, it will mess up some of the variables, and I'll ultimately have to go into the editor and change them back.